fe/fe-core/src/main/java/com/starrocks/common/MyTThreadPoolServer.java

```diff
@@ -0,0 +1,35 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.common;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.thrift.server.TThreadPoolServer;
+
+public class MyTThreadPoolServer extends TThreadPoolServer {
+    private static final Logger LOG = LogManager.getLogger(MyTThreadPoolServer.class);
+
+    public MyTThreadPoolServer(Args args) {
+        super(args);
+    }
+
+    /**
+     * override the execute method, catch any throwable and retry to avoid the accept thread exit unexpected
+     */
+    @Override
+    protected void execute() {
+        boolean shouldSleep = false;
+        while (true) {
+            try {
+                if (shouldSleep) {
+                    Thread.sleep(5000);
+                }
+                shouldSleep = true;
+
+                super.execute();
+            } catch (Throwable t) {
```

Comment:
I honestly think catching all the Throwable and retry is a bad practice. If an unrecoverable Error is thrown, there is no use in trying, neither can the program recover from exiting the process and recover by Supervised. I suggest you only catch what you meant to catch and leave the rest raised.

Reply:
1. If some actual memory leak occurs, this process will still neither serve well nor can it exit. I suggest you override the function that checks the number of connections, raising a self-defined exception instead of OOM Error.
2. I don't see how other throwables can lead to an exit. There is no `raise` nor `break` in the loop.

Reply:
1. The Exception occurred is `java.lang.OutOfMemoryError : unable to create new native Thread`, the reason is the limit of user max process is reached, not the connection number. The exception thrown by JVM is a little confused, because it is not the memory problem. I will handle the exception according to the error message.  Only retry on `unable to create new native Thread`.